pkg/kf/commands/routes/integration_test.go

```diff
@@ -29,32 +30,60 @@ import (
 func TestIntegration_Routes(t *testing.T) {
 	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
 		hostname := fmt.Sprintf("some-host-%d", time.Now().UnixNano())
+		domain := "example.com"
+		path := "some-path"
 
-		findRoute := func(shouldFind bool) {
-			RetryOnPanic(ctx, t, func() {
-				var found bool
-				for _, line := range kf.Routes(ctx) {
-					expected := hostname + " example.com /some-path"
-					actual := strings.Join(strings.Fields(line), " ")
-					if expected == actual {
-						found = true
-						break
-					}
-				}
-
-				if shouldFind != found {
-					// We'll panic so we can use our retry logic
-					panic(fmt.Errorf("Wanted %v, got %v", shouldFind, found))
-				}
-			})
-		}
+		// TODO: use the domain from the cluster.
+		kf.CreateRoute(ctx, domain, "--hostname="+hostname, "--path="+path)
+		findRoute(ctx, t, kf, hostname, domain, path, true)
+		kf.DeleteRoute(ctx, domain, "--hostname="+hostname, "--path="+path)
+		findRoute(ctx, t, kf, hostname, domain, path, false)
+	})
+}
 
-		kf.Target(ctx, "default")
+// TestIntegration_UnmappedRoute creates a route via `create-route` that is not mapped to an app.
+// The test verifies that the route exists with `routes`, and checks that hitting the route returns a 503
+// with `proxy-route`.
+func TestIntegration_UnmappedRoute(t *testing.T) {
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		hostname := fmt.Sprintf("some-host-%d", time.Now().UnixNano())
+		domain := "example.com"
+		path := "mypath"
 
 		// TODO: use the domain from the cluster.
-		kf.CreateRoute(ctx, "example.com", "--hostname="+hostname, "--path=some-path")
-		findRoute(true)
-		kf.DeleteRoute(ctx, "example.com", "--hostname="+hostname, "--path=some-path")
-		findRoute(false)
+		kf.CreateRoute(ctx, domain, "--hostname="+hostname, "--path="+path)
+		routeHost := fmt.Sprintf("%s.%s", hostname, domain)
+		findRoute(ctx, t, kf, hostname, domain, path, true)
+
+		go kf.ProxyRoute(ctx, routeHost, 8083)
+
+		{
+			resp, respCancel := RetryGet(ctx, t, "http://localhost:8083/"+path, 90*time.Second, http.StatusServiceUnavailable)
+			defer resp.Body.Close()
+			defer respCancel()
+			Logf(t, "testing for 503")
+		}
+
+		kf.DeleteRoute(ctx, "example.com", "--hostname="+hostname, "--path="+path)
+		findRoute(ctx, t, kf, hostname, domain, path, false)
+	})
+}
+
+func findRoute(ctx context.Context, t *testing.T, kf *Kf, hostname, domain, path string, shouldFind bool) {
+	RetryOnPanic(ctx, t, func() {
```

Comment:
This should be changed to a normal control-flow operation, `RetryOnPanic` is an anti-pattern because it's using panics for flow-control

Reply:
Created and referenced an issue for this and the other RetryOnPanic in the app integration tests (https://github.com/google/kf/issues/699). In this case this would be solved once routes can be created synchronously